### ISSUE TITLE
immutable access to RawTable?

### ISSUE THREAD

**Issue by tower120** [rust-lang/hashbrown#403]
Why there is no immutable access to RawTable both in HashMap and HashSet?

For example, I want to emulate absent `HashSet::raw_entry().from_hash...` with `HashSet::raw_table().get()`
but there is only mutable access to underlying RawTable in both HashSet and HashMap.

And is there a reason why there is no `raw_entry` API for HashSet?

**Comment by Amanieu** [comment]
This is an oversight. We should probably have both `raw_table` and `raw_table_mut` methods.

There is no `raw_entry` API for HashSet simply because nobody has implemented one yet. But then again there are concerns about the raw entry API and whether we should keep it at all.

**Comment by tower120** [comment]
OK. I can implement `raw_table` + `raw_table_mut` and PR it - that's not problem.

Wouldn't it be a breaking change?
Maybe something like `raw_table_immut` instead?

**Comment by Amanieu** [comment]
The raw API isn't very stable, it's fine to make breaking changes. We'll just make another major version bump.

**Comment by tower120** [comment]
So can I do it now? Will you review/accept that PR?

**Comment by Amanieu** [comment]
Yes, but it may take a while until we actually publish a new release. I want to get all of the breaking changes in at once.

**Comment by tower120** [comment]
OK - sure. I'm starting the implementation.

**Comment by Amanieu** [comment]
Closing this issue since the raw table API has been removed.
